src/Claims.ts

```diff
@@ -230,6 +264,104 @@ export class Claims {
+   *
+   * @param opts.targets - Identities (or Identity IDs) for which to fetch targeting claims. Defaults to all targets
+   * @param opts.trustedClaimIssuers - Identity IDs of claim issuers. Defaults to all claim issuers
+   * @param opts.scope - scope of the claims to fetch. Defaults to any scope
+   * @param opts.claimTypes - types of the claims to fetch. Defaults to any type
+   * @param opts.includeExpired - whether to include expired claims. Defaults to true
+   * @param opts.size - page size
+   * @param opts.start - page offset
+   *
+   * @note supports pagination
+   * @note uses the middleware V2
+   */
+  public async getIdentitiesWithClaimsV2(
+    opts: {
+      targets?: (string | Identity)[];
+      trustedClaimIssuers?: (string | Identity)[];
+      scope?: Scope;
+      claimTypes?: Exclude<ClaimType, ClaimType.InvestorUniquenessV2>[];
+      includeExpired?: boolean;
+      size?: BigNumber;
+      start?: BigNumber;
+    } = {}
+  ): Promise<ResultSet<IdentityWithClaims>> {
+    const { context } = this;
+
+    const {
+      targets,
+      trustedClaimIssuers,
+      scope,
+      claimTypes,
+      includeExpired = true,
+      size = new BigNumber(DEFAULT_GQL_PAGE_SIZE),
+      start = new BigNumber(0),
+    } = opts;
+
+    let targetIssuers;
+
+    const filters = {
+      scope: scope ? scopeToMiddlewareScope(scope) : undefined,
+      trustedClaimIssuers: trustedClaimIssuers?.map(trustedClaimIssuer =>
+        signerToString(trustedClaimIssuer)
+      ),
+      claimTypes: claimTypes?.map(ct => ClaimTypeEnum[ct]),
+      includeExpired,
+    };
+
+    if (!targets) {
+      const {
+        data: {
+          claims: { groupedAggregates: groupedTargets },
+        },
+      } = await context.queryMiddlewareV2<EnsuredV2<QueryV2, 'claims'>>(
+        claimsGroupingQuery({
+          ...filters,
+        })
+      );
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      targetIssuers = flatten(groupedTargets!.map(groupedTarget => groupedTarget.keys!));
+    } else {
+      targetIssuers = targets.map(target => signerToString(target));
+    }
+
+    // note: pagination count is based on the target issuers and not the claims count
+    const count = new BigNumber(targetIssuers.length);
+
+    // tooling-gql does pagination based on sorted target issuers, hence the explicit `sort()` function (as graphql doesn't sort the final data)
+    targetIssuers.sort();
+    targetIssuers = targetIssuers.slice(start.toNumber(), size.plus(start).toNumber());
+
+    const {
+      data: {
+        claims: { nodes },
+      },
+    } = await context.queryMiddlewareV2<EnsuredV2<QueryV2, 'claims'>>(
+      claimsQuery({
+        dids: targetIssuers,
+        ...filters,
+      })
+    );
+
+    const data = toIdentityWithClaimsArrayV2(
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      nodes.map(node => node!),
```

Comment:
The problem is that the code is misleading. We were using an assertion because we know those values cannot be null, and the code is very explicit about that. Someone reading the code with `filter` will assume that the values *can* be null and we're ignoring them

Reply:
The "correct" way to handle this would be to remove the `Maybe`s from properties we know are always defined

Reply:
If the type says Maybe, then I will think its nullable, and not because of the filter. My concern is something changing and TS will say its OK and we get a runtime exception.

If we can get rid of the `Maybe` where possible I agree that would be better though.